src/Microsoft.DotNet.Wpf/src/Shared/System/Windows/Markup/TypeConverterHelper.cs

```diff
@@ -455,10 +452,14 @@ internal static TypeConverter GetCoreConverterFromCustomType(Type type)
         /// <returns>A TypeConverter for the Type type if found. Null otherwise.</returns>
         internal static TypeConverter GetTypeConverter(Type type)
         {
+#if SYSTEM_XAML
+            Debug.Assert(type != null, "Null passed for type to GetTypeConverter");
+#else
             if (type == null)
             {
                 throw new ArgumentNullException(nameof(type));
```

Comment:
What does it mean now that System.Xaml will no longer throw this exception?

Reply:
This is dead code in `System.Xaml.csproj` and is never reached. I didn't remove the code as it is shared. This way we explicitly acknowledge the dead code, but don't break anyone. For reasoning as to why it is dead, see description of PR

Reply:
Hi - backed out this change